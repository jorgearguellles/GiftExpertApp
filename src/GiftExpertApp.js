import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GiftGrid } from "./components/GiftGrid";

const GiftExpertApp = () => {
  
  const [categories, setCategories] = useState(['disney'])

  // const handleAdd = () =>{
  //   const newElement = "Meta";
  //   const adding = categories => [...categories, newElement];
  //   setCategories(adding)
  // };

  return(
    <>
      <h1>GiftExpertApp</h1>
      <AddCategory setCategories={setCategories} />
      <hr />
      <ol>
        {
          categories.map( category => (
              <GiftGrid 
                category={category}
                key={category}
              />
            )
          )
        }
      </ol>
    </>
  );
};

export {GiftExpertApp};