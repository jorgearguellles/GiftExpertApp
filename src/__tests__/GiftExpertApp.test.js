import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import { GiftExpertApp } from '../GiftExpertApp';

describe("<GiftExpertApp /> testing", ()=>{

  test("1. Should render correctly", ()=>{
    const wrapper = shallow(<GiftExpertApp />)
    expect(wrapper).toMatchSnapshot();
  });
})