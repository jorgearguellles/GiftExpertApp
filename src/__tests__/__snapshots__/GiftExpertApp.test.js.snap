// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GiftExpertApp /> testing 1. Should render correctly 1`] = `
<Fragment>
  <h1>
    GiftExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GiftGrid
      category="disney"
      key="disney"
    />
  </ol>
</Fragment>
`;
